spyder/utils/environ.py

```diff
@@ -97,90 +120,57 @@ def __init__(self, environ, parent=None):
-                        _("Module <b>pywin32 was not found</b>.<br>"
-                          "Please restart this Windows <i>session</i> "
-                          "(not the computer) for changes to take effect."))
-
-    class WinUserEnvDialog(CollectionsEditor):
-        """Windows User Environment Variables Editor"""
-        def __init__(self, parent=None):
-            super(WinUserEnvDialog, self).__init__(parent)
-            self.setup(get_user_env(),
-                       title=r"HKEY_CURRENT_USER\Environment")
+class UserEnvDialog(RemoteEnvDialog):
+    """User Environment Variables Viewer/Editor"""
+
+    def __init__(self, parent=None):
+        title = "User Environment variables"
+        readonly = True
+        if os.name == 'nt':
+            title = r"HKEY_CURRENT_USER\Environment"
+            readonly = False
+
+        super(UserEnvDialog, self).__init__(get_user_env(), parent,
+                                            title, readonly)
+
+        if os.name == 'nt':
             if parent is None:
                 parent = self
-            QMessageBox.warning(parent, _("Warning"),
-                        _("If you accept changes, "
-                          "this will modify the current user environment "
-                          "variables directly <b>in Windows registry</b>. "
-                          "Use it with precautions, at your own risks.<br>"
-                          "<br>Note that for changes to take effect, you will "
-                          "need to restart the parent process of this applica"
-                          "tion (simply restart Spyder if you have executed it "
-                          "from a Windows shortcut, otherwise restart any "
-                          "application from which you may have executed it, "
-                          "like <i>Python(x,y) Home</i> for example)"))
-
-        def accept(self):
-            """Reimplement Qt method"""
+            QMessageBox.warning(
+                parent, _("Warning"),
+                _("If you accept changes, "
+                  "this will modify the current user environment "
+                  "variables directly <b>in Windows registry</b>. "
+                  "Use it with precautions, at your own risks.<br>"
+                  "<br>Note that for changes to take effect, you will "
+                  "need to restart the parent process of this applica"
+                  "tion (simply restart Spyder if you have executed it "
+                  "from a Windows shortcut, otherwise restart any "
+                  "application from which you may have executed it, "
+                  "like <i>Python(x,y) Home</i> for example)")
+            )
+
+    def accept(self):
+        """Reimplement Qt method"""
+        if os.name == 'nt':
             set_user_env(listdict2envdict(self.get_value()), parent=self)
-            QDialog.accept(self)
+        super(UserEnvDialog, self).accept()
```

Comment:
```suggestion
        super().accept()
```

This is the new, Python 3 way of using `super`.